### ISSUE TITLE
Handle solver_params.xml

### ISSUE THREAD

**Issue by merijn** [nlesc-smcm/i-emic#65]
Currently `Ocean`, `CoupledModel` and `Topo` independently load `solver_params.xml`. The problem with this situation is that Ocean's parameters are split over two (well, three, see discussion of THCM in #64) parameter lists. Initially I was going to make the solver params a sublist of the `Ocean` parameters so that `CoupledModel` and `Topo` could just query them from the underlying model. This doesn't work directly, since `CoupledModel` and `Topo` would inherit the default parameter values of `Ocean` which don't match their current default solver parameters.

The alternative solution is to duplicate the current `solver_params.xml` contents to the parameter lists of `Ocean`, `CoupledModel`, and `Topo`, but this only makes sense if the values don't have to be kept in sync across these.

**Comment by Sbte** [comment]
I suggest actual parameter lists are only loaded from the main files, and then indeed combined into one with sublists.

`Ocean`, `CoupledModel` and `Topo` should only use a solver when being run as "main" problem. So when `CoupledModel` is used, `Ocean` does not use its solver (as far as I know). So no need for synchronization or duplication.

**Comment by merijn** [comment]
@Sbte I ran into an issue where `test_topo` lists several "Topo" parameters for the solver, but those don't seem to actually ever get passed to the solver? That doesn't seem right...

**Comment by Sbte** [comment]
What do you mean exactly? The ones I found get passed to the solver in `topo/Topo.H`

**Comment by merijn** [comment]
Ah! I was just looking at the Ocean part of things, I forgot that `solver_params.xml` also gets used by other parts of the code.

**Comment by Sbte** [comment]
Well, since you're implementing it with sublists now, you can make it a Topo -> Solver sublist, and get rid of the "Topo " part of the parameter name. That looks like a much nicer way of doing it.

**Comment by merijn** [comment]
Yeah, I'll do that and have both Topo and Ocean read the same solver_params.xml (since they're currently the same), if someone ever needs a different config they just need to read in a different file for Topo in the test/code.

**Comment by Sbte** [comment]
I'm already getting excited about the huge number of lines that you can remove in the near future.

**Comment by Sbte** [comment]
Mostly fixed by 9bec0ebda0ccb829c06b915e43601518ed4fd3bc, but still has to be fixed in CoupledModel.
